File: hosung/leetcode/378. Kth Smallest Element in a Sorted Matrix.py
class Solution(object):
    def kthSmallest(self, matrix, k):
        a = []
        b = len(matrix)
        c = len(matrix[0])
        for i in range(b):
            for j in range(c):
                a.append(matrix[i][j])
        a.sort()
        return a[k-1]

File: taejoong/leetcode/378_Kth-Smallest-Element-in-a-Sorted-Matrix.java
class Solution {
    public int kthSmallest(int[][] matrix, int k) {

        PriorityQueue<Integer> pq = new PriorityQueue<>();

        for(int[] nums : matrix) {
            for(int num : nums) {
                pq.offer(num);
            }
        }

        for(int i = 1; i < k; i++) {
            pq.poll();
        }

        return pq.peek();
        
    }
}